### PR TITLE
FIX: get  rotation for node with skew.

### PR DESCRIPTION
**Pull request by DefaultCyberid**
Error: RotationX != RotationY. Don't know which one to return.

### CODE REVIEW

File: cocos2dx-lua/poco/Cocos2dxNode.lua [hunk 1]
```diff
@@ -166,10 +166,12 @@ function Node:getAttr(attrName)
...
 
     elseif attrName == 'rotation' then
-        if self.node.getRotation ~= nil then
-            return self.node:getRotation()
+        local rotationX, rotationY
+        if self.node.getRotationSkewX ~= nil and self.node.getRotationSkewY ~= nil then
+            rotationX, rotationY = self.node:getRotationSkewX(), self.node:getRotationSkewY()
         end
-        return nil
+        return (rotationX == rotationY) and rotationX or {rotationX, rotationY}
```
Comment (adolli) [inline cocos2dx-lua/poco/Cocos2dxNode.lua]: I think the attribute rotation should be defined as determined value type. Is there a way to return the rotation value by following sequence `{rotationQant}` -> `{rotation3D, 0}` -> `{rotationSkewX, rotationSkewY, 0, 0}` -> `{rotationSkewX, 0, 0, 0}`, which I looked up some [Cocos2d-x Node Api reference here](http://api.cocos.org/d3/d82/classcocos2d_1_1_node.html#af1ab5fd7e4320c19af50fc3ddf055bc2).

**Comment by adolli** [comment]
Thank you for your PR and it really solves the bug which bothered me for a long time. I left some review message for you. Please take a look.

File: cocos2dx-lua/poco/Cocos2dxNode.lua (hunk 1 above)
Comment (DefaultCyberid) [reply cocos2dx-lua/poco/Cocos2dxNode.lua]: I don't think it a good idea. Quaternion, rotation3D and rotational skew are different.
I can put rotationSkewX and rotationSkewY together no matter if they are equal, but it seems weird to write "rotation = 30" as "rotation = {30, 30}". What do you think?

Comment (adolli) [reply cocos2dx-lua/poco/Cocos2dxNode.lua]: In 2D space, the rotation is only a scaler. While in 3D, rotation in space must be defined by Quaternion. If you treat cocos2dx as a 2D game engine, you can just define the rotation as `rotationSkewX` or just the `default rotation`. And also add another attribute in subclass like "rotationQuat" to return the `{rotationQant}`, if you wish.

Comment (adolli) [reply cocos2dx-lua/poco/Cocos2dxNode.lua]: So generally, is that simply returning the `getRotationSkewX()` meets your needs?

Comment (DefaultCyberid) [reply cocos2dx-lua/poco/Cocos2dxNode.lua]: You're right, I'll add field rotation3D and skew.

**Comment by DefaultCyberid** [comment]
Field rotation3D ( shown while rotationSkewX == rotationSkewY ) and skew added.
Quaternion is unavailable in lua.

File: cocos2dx-lua/poco/Cocos2dxNode.lua [hunk 2]
```diff
@@ -166,10 +168,27 @@ function Node:getAttr(attrName)
         return self.node:getDescription()
 
     elseif attrName == 'rotation' then
-        if self.node.getRotation ~= nil then
-            return self.node:getRotation()
+        local rotationX, rotationY
+        if self.node.getRotationSkewX ~= nil and self.node.getRotationSkewY ~= nil then
+            rotationX, rotationY = self.node:getRotationSkewX(), self.node:getRotationSkewY()
+        end
+        return (rotationX == rotationY) and rotationX or {rotationX, rotationY}
```
Comment (adolli) [inline cocos2dx-lua/poco/Cocos2dxNode.lua]: in "rotation" I am going to simply return the `rotationX`

**Comment by adolli** [comment]
@DefaultCyberid thank u so much! I am going to have it tested.

File: cocos2dx-lua/poco/Cocos2dxNode.lua (hunk 2 above)
Comment (DefaultCyberid) [reply cocos2dx-lua/poco/Cocos2dxNode.lua]: They are equal only when rotationX == rotationY.
Add another field named "rotational skew" that returns rotationX and rotationY, set "rotation" hidden or zero when they are not equal, that's a better idea.

BTW, Standard skew (getSkewX/Y) and rotational skew (getRotationSkewX/Y) are different.
https://www.youtube.com/watch?v=G6DfZr0zYnw

Comment (adolli) [reply cocos2dx-lua/poco/Cocos2dxNode.lua]: you are right. I just want to make the return value type to be only one type. I someone want to get rotation skew X and Y, they can call another attribute.
the rotational skew looks like rotate an object in 3D space. I'll have this attribute updated.

**Pull request merged**